app/src/main/java/com/github/se/polyfit/ui/components/AddIngredientPopup.kt

```diff
@@ -0,0 +1,197 @@
+      leadingIcon = {
+        Icon(Icons.Filled.Search, contentDescription = "search", tint = SecondaryGrey)
+      },
+      placeholder = {
+        Text(
+            text = "Find an Ingredient...",
+            color = SecondaryGrey,
+            style = TextStyle(fontSize = 17.sp))
+      }) {
+        LazyColumn(modifier = Modifier.testTag("IngredientSearchScrollableList")) {
+          items(filteredIngredients) { ingredient ->
+            Text(
+                text = ingredient,
+                modifier =
+                    Modifier.testTag("SearchResult $ingredient")
+                        .fillMaxWidth()
+                        .clickable {
+                          searchText = ingredient
+                          showIngredientSearch.value = false
+                          // TODO: Apply according nutrition facts to nutrition text fields
+                        }
+                        .padding(8.dp))
+            Divider()
+          }
+        }
+      }
+}
+
+@Composable
+fun AddButton(onClick: () -> Unit, modifier: Modifier) {
+  Button(
+      onClick = onClick,
+      modifier = modifier.padding(top = 16.dp).fillMaxWidth(0.5f),
+      shape = RoundedCornerShape(50.dp),
+      colors = ButtonDefaults.buttonColors(containerColor = PrimaryPurple)) {
+        Text("Add", color = Color.White)
+      }
+}
+
+@Composable
+fun AddIngredientDialog(onClickCloseDialog: () -> Unit) {
+  Dialog(onDismissRequest = onClickCloseDialog) {
+    GradientBox(
+        outerModifier = Modifier.testTag("AddIngredientPopupContainer"),
+        innerModifier = Modifier.fillMaxWidth(),
+        iconOnClick = onClickCloseDialog,
+        icon = Icons.Filled.Close,
+        iconColor = PrimaryPink,
+        iconDescriptor = "close",
+    ) {
+      Column(
+          modifier =
+              Modifier.fillMaxWidth()
+                  .padding(top = 20.dp, bottom = 20.dp)
+                  .testTag("AddIngredientContentContainer")) {
+            Spacer(modifier = Modifier.height(16.dp))
+
+            SearchIngredients()
+
+            Spacer(modifier = Modifier.height(16.dp))
+
+            NutritionFacts()
+
+            AddButton(
+                onClick = {
+                  // TODO: Add ingredient to ingredient list
+                  onClickCloseDialog()
+                },
+                modifier =
+                    Modifier.testTag("AddIngredientToListButton")
+                        .align(Alignment.CenterHorizontally))
+          }
+    }
+  }
+}
+
+// Temporary function for testing purpose
+// @Composable
+// @Preview
+// fun TestPopupWindow() {
+//  val showAddIngredDialog = remember { mutableStateOf(true) }
+//
+//  Button(onClick = { showAddIngredDialog.value = true }) { Text("Show Pop-Up") }
+//
+//  if (showAddIngredDialog.value) {
+//    AddIngredientDialog(onClickCloseDialog = { showAddIngredDialog.value = false })
+//  }
+//    AddIngredientDialog(onClickCloseDialog = { })
+// }
```

Comment:
```suggestion

```

Reply:
will delete when feature is ready to be merged, as this is used for previewing what the result should look like